src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SealInternalTypes.cs

```diff
@@ -39,20 +39,21 @@ public override void Initialize(AnalysisContext context)
 
         private static void OnCompilationStart(CompilationStartAnalysisContext context)
         {
-            // To avoid false positives, as with CA1812 (avoid uninstantiated internal classes), skip any assemblies with InternalsVisibleTo.
-            var internalsVisibleToAttributeSymbol = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeCompilerServicesInternalsVisibleToAttribute);
-            if (context.Compilation.Assembly.HasAttribute(internalsVisibleToAttributeSymbol))
-            {
-                return;
-            }
-
             INamedTypeSymbol? comImportAttributeType = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesComImportAttribute);
 
             var candidateTypes = PooledConcurrentSet<INamedTypeSymbol>.GetInstance(SymbolEqualityComparer.Default);
             var baseTypes = PooledConcurrentSet<INamedTypeSymbol>.GetInstance(SymbolEqualityComparer.Default);
+            var hasInternalsVisibleTo = context.Compilation.Assembly.HasAttribute(
+                context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeCompilerServicesInternalsVisibleToAttribute));
 
             context.RegisterSymbolAction(context =>
             {
+                if (hasInternalsVisibleTo
```

Comment:
Same suggestion as before - delay the option check to as late as possible.

Reply:
Done.